File: Selenium/Activities/Activity7.py
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:

    # Open the link in browser
    driver.get("https://training-support.net/selenium/drag-drop")

    # Print page title
    print("Page Title is: " + driver.title)

    # Identify ball and dropzones
    football = driver.find_element(By.ID, "draggable")
    dragzone1 = driver.find_element(By.ID, "droppable")
    dragzone2 = driver.find_element(By.ID, "dropzone2")

    # perform drag and drop
    actions = webdriver.ActionChains(driver)

    actions.drag_and_drop(football, dragzone1).perform()
    message = driver.find_element(By.XPATH, "//div[@id='droppable']/p").text
    print(message)

    actions.drag_and_drop(football, dragzone2).perform()
    message = driver.find_element(By.XPATH, "//div[@id='dropzone2']/p").text
    print(message)

    driver.quit();

File: Selenium/Activities/Activity7.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

import javax.swing.*;

public class Activity7 {
    public static void main(String[] args) throws InterruptedException {
        // Installing Firefox driver
        WebDriverManager.firefoxdriver().setup();

        // Suppressing browser log
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "NULL");
        // Initialize firefox driver
        WebDriver driver = new FirefoxDriver();
        Actions builder = new Actions(driver);

        // Open a page
        driver.get("https://training-support.net/selenium/drag-drop");
        System.out.println(driver.getTitle());

        //Identify ball and dropzones
        WebElement football = driver.findElement(By.id("draggable"));
        WebElement dragzone1 = driver.findElement(By.id("droppable"));
        WebElement dragzone2 = driver.findElement(By.id("dropzone2"));

        //perform drag and drop
        builder.dragAndDrop(football, dragzone1).build().perform();
        String message = driver.findElement(By.xpath("//div[@id='droppable']/p")).getText();
        System.out.println(message);

        builder.dragAndDrop(football, dragzone2).build().perform();
        message = driver.findElement(By.xpath("//div[@id='dropzone2']/p")).getText();
        System.out.println(message);

        driver.quit();
    }
}
